paddle/fluid/framework/ir/mkldnn/params_quantization_mkldnn_pass.cc

```diff
@@ -0,0 +1,219 @@
+
+  conv_op->Op()->SetAttr("Bias_scales", std::vector<float>(1, 1));
+}
+
+}  // namespace
+
+ParamsQuantizationMkldnnPass::ParamsQuantizationMkldnnPass() {
+  AddOpCompat(OpCompat("conv2d"))
+      .AddInput("Input")
+      .IsTensor()
+      .End()
+      .AddInput("Filter")
+      .IsTensor()
+      .End()
+      .AddInput("Bias")
+      .IsTensor()
+      .IsOptional()
+      .End()
+      .AddInput("ResidualData")
+      .IsTensor()
+      .IsOptional()
+      .End()
+      .AddOutput("Output")
+      .IsTensor()
+      .End()
+      .AddAttr("strides")
+      .IsType<std::vector<int>>()
+      .End()
+      .AddAttr("paddings")
+      .IsType<std::vector<int>>()
+      .End()
+      .AddAttr("padding_algorithm")
+      .IsOptional()
+      .IsStringIn({"EXPLICIT", "SAME", "VALID"})
+      .End()
+      .AddAttr("groups")
+      .IsNumGE(1)
+      .End()
+      .AddAttr("dilations")
+      .IsType<std::vector<int>>()
+      .End()
+      .AddAttr("data_format")
+      .IsStringIn({"NCHW", "AnyLayout"})
+      .End();
+}
+
+void ParamsQuantizationMkldnnPass::Conv(ir::Graph* graph,
+                                        bool with_residual_data) const {
+  GraphPatternDetector gpd;
+  patterns::ConvResidual conv_pattern(gpd.mutable_pattern(), name_scope_);
+  conv_pattern(with_residual_data);
+
+  int params_to_int8_conv_found = 0;
+
+  auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
+                     Graph* g) {
+    if (!IsCompat(subgraph, g)) {
+      LOG(WARNING) << "Pass in op compat failed.";
+      return;
+    }
+    VLOG(4) << "handle convolution in params_quantization_mkldnn_pass";
+
+    GET_IR_NODE_FROM_SUBGRAPH(conv_op, conv_op, conv_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(conv_input, conv_input, conv_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(conv_filter, conv_filter, conv_pattern);
+
+    // get scope to interact with tensors
+    auto* scope = param_scope();
+    PADDLE_ENFORCE_NOT_NULL(
+        scope, platform::errors::InvalidArgument("Scope cannot be nullptr."));
+
+    if (ShouldSkipConv(conv_op, scope, conv_filter)) {
+      return;
+    }
+
+    QuantizeConvFilter(scope, g, conv_op, conv_filter);
+
+    if (HasBias(conv_op)) {
+      QuantizeConvBias(scope, g, conv_op);
+    }
+    params_to_int8_conv_found++;
+  };
+  gpd(graph, handler);
+  AddStatis(params_to_int8_conv_found);
+
+  std::stringstream msg_ss;
+  msg_ss << "Quantized params of " << params_to_int8_conv_found
+         << " conv2d ops";
+  if (with_residual_data) msg_ss << " with residual connection";
+  paddle::string::PrettyLogDetail(msg_ss.str().c_str());
+}
+
+void ParamsQuantizationMkldnnPass::ApplyImpl(ir::Graph* graph) const {
+  PADDLE_ENFORCE_NOT_NULL(graph,
+                          platform::errors::InvalidArgument(
+                              "Pointer to graph argument should not be NULL."));
+  FusePassBase::Init(name_scope_, graph);
+  Conv(graph, true);
+  Conv(graph, false);
```

Comment:
create a helper variables for these flags, or enum.
It looks like magic right now.

Reply:
What do you mean or suggest here?

Reply:
Wouldn't it look weird with enum? In future there will be more function calls here. I got inspiration from here
https://github.com/PaddlePaddle/Paddle/blob/19a7524febaf732108d6faad26724b265b56c298/paddle/fluid/framework/ir/mkldnn/cpu_quantize_pass.cc#L1181-L1197

Reply:
The easiest way to fix it for now is to create a variable `auto with_residual_data = true | false` and then pass it into this function.
Don't use bare bools, opaque it with variables.